script/v2/fork-scenarios/scUSDCv2Rebalance.s.sol

```diff
@@ -0,0 +1,216 @@
+        vm.warp(block.timestamp + 365 days);
+        console2.log("\n-- ltvs changed after 356 days --");
+        _assertLtvsAreAboveTarget();
+        _logVaultInfo();
+
+        // 2. rebalance to the target ltv
+        _rebalance();
+        console2.log("\n-- rebalance to get ltvs to target values --");
+        _assertLtvsAreAtTarget();
+        _logVaultInfo();
+
+        vm.stopBroadcast();
+
+        // make a deposit of 200k USDC from bob
+        _depositUsdc(200_000e6, bob);
+
+        // 3. rebalance to reinvest the additional deposit
+        vm.startBroadcast(keeper);
+        console2.log("\n-- reinvest additional deposits --");
+        _invest();
+        _assertLtvsAreAtTarget();
+        _logVaultInfo();
+        vm.stopBroadcast();
+
+        // add 10% to the scWETH vault to simulate profits from staking
+        ERC20 weth = scWethV2.asset();
+        deal(address(weth), address(scWethV2), weth.balanceOf(address(scWethV2)).mulWadUp(1.1e18));
+
+        // 4. sell profit and reinvest
+        vm.startBroadcast(keeper);
+        console2.log("\n-- sell & reinvest profits --");
+        // note: selling profits can be done as part of the rebalance call (added to multicall data) but usdc received from selling will has to be estimated in that case
+        scUsdcV2.sellProfit(0); // _usdcAmountOutMin = 0
+        _invest();
+        _assertLtvsAreAtTarget();
+        _logVaultInfo();
+    }
+
+    function _fork(uint256 _blockNumber) internal {
+        uint256 mainnetFork = vm.createFork(vm.envString("RPC_URL_MAINNET"));
+        vm.selectFork(mainnetFork);
+        vm.rollFork(_blockNumber);
+    }
+
+    function _deployVaults() internal {
+        vm.startBroadcast(deployerAddress);
+
+        Swapper swapper = new Swapper();
+        priceConverter = new PriceConverter(deployerAddress);
+
+        scWethV2 = new scWETHv2(deployerAddress, keeper, 0.99e18, weth, swapper, priceConverter);
+        scUsdcV2 = new scUSDCv2(deployerAddress, keeper, scWethV2, priceConverter, swapper);
+
+        aaveV2Adapter = new AaveV2ScUsdcAdapter();
+        aaveV3Adapter = new AaveV3ScUsdcAdapter();
+
+        scUsdcV2.addAdapter(aaveV2Adapter);
+        scUsdcV2.addAdapter(aaveV3Adapter);
+
+        vm.stopBroadcast();
+    }
+
+    function _depositUsdc(uint256 _depositAmount, address _account) internal {
+        deal(address(scUsdcV2.asset()), _account, _depositAmount);
+
+        vm.startPrank(_account);
+        scUsdcV2.asset().approve(address(scUsdcV2), _depositAmount);
+        scUsdcV2.deposit(_depositAmount, _account);
+        vm.stopPrank();
+
+        console2.log("\ndeposited %s from %s", _depositAmount, _account);
+    }
+
+    function _invest() internal {
+        uint256 minFloatRequired = scUsdcV2.totalAssets().mulWadUp(scUsdcV2.floatPercentage());
+        uint256 investableAmount = scUsdcV2.usdcBalance() - minFloatRequired;
+
+        uint256 aaveV3Collaateral = investableAmount.mulWadDown(aaveV3AllocationPercent);
```

Comment:
can we add comment for mulWadDown

Reply:
not sure what exactly do you mean.

Reply:
it's just not clear why that calculation equals the collateral

also there is a typo there

`...Collaateral` should be `...Collateral`

Reply:
mulWadDown multiplies investableAmount with aaveV3AllocationPercent and then divides the result by 1e18. 
So its like multiplying the amount with a percent value.

For e.g,
investableAmount = 12e18
aaveV3AllocationPercent = 0.5e18 (this is a percent value with 1e18 = 100% so it must always be less than 1e18)
then, the result aaveV3Collateral will be = (12e18*0.5e18) / 1e18 = 6e18

Its similar to 12 * 0.5 = 6